kernel/libs/aster-bigtcp/src/iface/common.rs

```diff
@@ -273,11 +297,49 @@ impl<E: Ext> BoundPort<E> {
         };
         Some(IpEndpoint::new(ip_addr, self.port))
     }
+
+    /// Marks the specified port as reusable.
+    pub fn set_can_reuse(&self, can_reuse: bool) {
+        if self.can_reuse.load(Ordering::Relaxed) == can_reuse {
```

Comment:
To avoid race condition, this check must be performed after holding the lock:
```rust
        let iface_common = self.iface.common();
        let mut used_ports = iface_common.used_ports.lock();

        if self.can_reuse.load(Ordering::Relaxed) == can_reuse {
            return;
        }

        // ...
```